src/current/v23.1/advanced-changefeed-configuration.md

```diff
@@ -33,6 +33,39 @@ kvadmission.rangefeed_catchup_scan_elastic_control.enabled
+
+To improve changefeed latency, you can use the following [cluster settings]({% link {{ page.version.version }}/cluster-settings.md %}) to reduce bursts of rangefeed work so that updates are paced steadily over time. It is necessary to adjust these cluster settings together in order to have an effect on changefeed latency. However, note that these can have an effect on running transactions, and should be adjusted and tested with your particular workload.
+
+#### `kv.closed_timestamp_target_duration`
+
+Each [range]({% link {{ page.version.version }}/architecture/overview.md %}#range) in CockroachDB tracks a property called the [_closed timestamp_]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#closed-timestamps), which means that no new writes can ever be introduced at or below that timestamp. Decreasing the target duration can reduce changefeed latency, so that the changefeed job can [checkpoint]({% link {{ page.version.version }}/how-does-an-enterprise-changefeed-work.md %}) more frequently and deliver messages to the sink.
+
+Default: `3s`
+Recommendation: Decreasing to `1s`–`2s`.
+
+{{site.data.alerts.callout_danger}}
+Thoroughly test any decrease to the closed timestamp interval before deploying the change in production, because such an adjustment can cause [contention]({% link {{ page.version.version }}/performance-best-practices-overview.md %}#understanding-and-avoiding-transaction-contention) between transactions.
+{{site.data.alerts.end}}
+
+#### `kv.closed_timestamp.side_transport_interval`
+
+When there is a period of no write activity, the _side-transport_ is a node-level process that fixes a timestamp to close for groups of ranges. This cluster setting adjusts how often the side-transport process will attempt to move the [closed timestamp]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#closed-timestamps) forward. The smaller the interval, the more frequently the side-transport will attempt to move the timestamp forward.
+
+This will also affect changefeed checkpoints. That is, if the interval is decreased the changefeed will be able to read and checkpoint more frequently, rather than collecting changes into fewer, less frequent checkpoints. To improve changefeed latency, we recommend decreasing the `kv.closed_timestamp.side_transport_interval` cluster setting.
+
+For more information on the side-transport interval, refer to [An epic read on follower reads](https://www.cockroachlabs.com/blog/follower-reads-stale-data/) on the Cockroach Labs Blog.
+
+#### `kv.rangefeed.closed_timestamp_smear_interval`
+
+This setting provides a mechanism to pace the [closed timestamp]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#closed-timestamps) notifications to follow replicas. Decreasing the closed timestamp smear interval makes rangefeed closed timestamp delivery less spiky, which can reduce its impact on foreground SQL query latency.
```

Comment:
I think we should mention that the default is 1ms; and that's reasonably.
It can be set to higher values, but anything higher than 20ms probably will not matter.

I also think we should expand this section to include an explanation of what this setting actually does.  Let's say you have a fairly large table, and one of the nodes in the cluster is hosting 5000 ranges from this table.  Normally, the the rangefeed system will wake up every `kv.closed_timestamp_target_duration` (say 3s) and every 3 seconds it'll publish checkpoints for all 6000 ranges.  What this setting says is: take 3s frequency and divide it into 1ms chunks -- there are 3000 of those chunks. And instead of publishing checkpoints for all 6000 ranges, publish checkpoints for 2 ranges -- but do this every 1 millisecond.

This sounds like a lot of work, but what this does is that it generates smooth and predictable load, instead of spiky, large (6000 ranges!) load.

Reply:
I have included your example here — I think this is a good illustration. What do you think?

Reply:
It's great.